### PR TITLE
- added unpushed code (csv, json, deep of inherithance tree)

### LINKED ISSUE #294: What happened with --report-json option?
The 2.x version of metrics does not have a super-useful option for furhter processing, that is, generating a JSON report with `--report-json` flag.

The HTML output is fancy, undoubtedly, but is hardly processable by another tools (like CI server, some stats calculator etc).

Why was it removed? Do I need some plugin for this?

### PR DESCRIPTION
**Pull request by Halleck45**
I'm sorry, that's a big PR :/

This PR contains my unpushed code:

Features

+ new metric : depth of inheritance tree
+ json report (fixes #294)
+ csv report

Fixes

+ fixes Deb artefact issues

Code

+ way to list all parents of class
+ way to get depth adnd childs of node
+ way to get average height of branches in graph

### CODE REVIEW

File: src/Hal/Metric/Registry.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            'nbMethodsSetters',
+            'afferentCoupling',
+            'efferentCoupling',
+            'instability',
+            'depthOfInheritanceTree',
+            'pageRank',
+
+        ];
+    }
+}
```
Comment (niconoe-) [inline src/Hal/Metric/Registry.php]: Missing EOL at end of file here.

File: tests/Component/Tree/GraphTest.php [hunk 2]
```diff
@@ -86,4 +86,36 @@ public function testEdgeIsAddedToFromAndToNode() {
...
+            'E' => false,
+        ];
+        foreach($roots as $node) {
+            $rootsFound[$node->getKey()] = true;
+        }
+
+        $this->assertTrue($rootsFound['A']);
+        $this->assertTrue($rootsFound['E']);
+    }
 }
```
Comment (niconoe-) [inline tests/Component/Tree/GraphTest.php]: Missing EOL at end of file here.

File: src/Hal/Application/Config/Validator.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function validate(Config $config)
         $config->set('exclude', array_filter(explode(',', $config->get('exclude'))));
 
         // parameters with values
-        $keys = ['report-html', 'report-violation', 'extensions'];
+        $keys = ['report-html', 'report-csv', 'report-violation', 'extensions'];
```
Comment (niconoe-) [inline src/Hal/Application/Config/Validator.php]: Key "report-json" seems to miss in this array... It that a wanted missing?

**Comment by niconoe-** [comment]
In a general point of view, this seems OK to me except for the 3 conversations opened.

But the consistency of the source code style in the files is very low and I think a big review with addition of such PHP_CodeSniffer tool would be a very good improvement to let more people working in this project.
Anyway, that is your decision to define the phpcs rules you want to apply and must definitly be part of a new PR.

EDIT: I saw you just commited fixtures about the 3 opened conversations, I'll be able to merge the PR asap.
Thanks.

**Comment by Halleck45** [comment]
+1 for adding a code style checking tool.

Thanks for your review

**Pull request merged**